polar-core/src/polar.rs

```diff
@@ -70,10 +75,58 @@ impl Polar {
         Self {
             kb: Arc::new(RwLock::new(KnowledgeBase::new())),
             messages: MessageQueue::new(),
+            loaded_content: Arc::new(RwLock::new(HashMap::new())),
+            loaded_files: Arc::new(RwLock::new(HashSet::new())),
         }
     }
 
-    pub fn load_file(&self, src: &str, filename: Option<String>) -> PolarResult<()> {
+    fn check_file(&self, src: &str, filename: &str) -> PolarResult<()> {
```

Comment:
Simply relying on the hashing algorithm used in `HashSet` and `HashMap` to check for collisions. This isn't guaranteed to be stable across different versions of rustc. Do we think this is a problem?

Reply:
What's the implication of this? These checks won't always work?

Reply:
I'm pretty sure its a non-issue. The checks wouldn't match if we somehow made it possible to do hot-swapped oso updates. But even then it's pretty nebulous.